interview-questions.md

```diff
@@ -8,36 +8,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 action: "GET" location: /students
+    This controller method would be index as a GET request would need to collect the data in students through the given location route. We know this method is probably index based on the fact that no params are specified in the given route which would imply that the active record query is probably .all which often lives in an index method of a given controller.
 
 action: "POST" location: /students
+    This route implies the use of the controller method create. The only differentiation from the above route that specifies the use of create is the fact that the given CRUD action is POST. POST is often used to create a new item which in this case will come from a shown form to add the new item.
 
 action: "GET" location: /students/new
+    This route is tied to the controller method new. We know this because of the location route /new. This shows the user will be redirect to a form in which they'll be able to apply information specified by a .erb file that the formatting for the given form. This will specify fields and given variables a user can apply information to be stored. Its important to note, without a create method, the data will not be applied to the database.
 
 action: "GET" location: /students/2
+    This route is indicative of the show method being the controller method used here. The show method typically will take the primary key from a given database element and return the element at that given key. In this case the 2 specified in the url will show the element with the primary key 2. 
 
 action: "GET" location: /students/2/edit
+    This route specifies the use of the edit method in the controller. The edit method takes a primary key and then directs the user to a form, specified by the edit keyword in the url. Here the url specifies that the database element with the primary key 2 is being called to be edited. Its important to note that these changes will only be saved if the controller contains a specified update method.
 
 action: "PATCH" location: /students/2
+    Here, in sequence and as mentioned earlier, the specified action implies that an update method is being used to apply the changes made to a given database element with a primary key of 2. We know this not only because of the primary key in the url but also because of the CRUD action PATCH, which calls for information to be updated.
 
 action: "DELETE" location: /students/2
+    The specified CRUD action here implies the use of the destroy method. We know this not only because of the DELETE command but also because of the specified key in the url.
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
```

Comment:
being mindful of the user experience really enhances the quality of an application